packages/fast-usdc/src/cli/transfer.js

```
@@ -0,0 +1,92 @@
+  queryForwardingAccount,
+  registerFwdAccount,
+} from '../util/noble.js';
+import { queryFastUSDCLocalChainAccount } from '../util/agoric.js';
+
+/** @typedef {import('../util/file').file} file */
+/** @typedef {import('@agoric/client-utils').VStorage} VStorage */
+/** @typedef {import('@cosmjs/stargate').SigningStargateClient} SigningStargateClient */
+/** @typedef {import('ethers').ethers.JsonRpcProvider} ethProvider */
+
+const transfer = async (
+  /** @type {file} */ configFile,
+  /** @type {string} */ amount,
+  /** @type {string} */ destination,
+  out = console,
+  fetch = globalThis.fetch,
+  /** @type {VStorage | undefined} */ vstorage,
+  /** @type {{signer: SigningStargateClient, address: string} | undefined} */ nobleSigner,
+  /** @type {ethProvider | undefined} */ ethProvider,
+) => {
+  const execute = async (
+    /** @type {import('./config').ConfigOpts} */ config,
+  ) => {
+    vstorage ||= makeVStorage(
+      { fetch },
+      { chainName: 'agoric', rpcAddrs: [config.agoricRpc] },
+    );
+    const agoricAddr = await queryFastUSDCLocalChainAccount(vstorage, out);
+    const appendedAddr = `${agoricAddr}+${destination}`;
+    out.log(`forwarding destination ${appendedAddr}`);
+
+    const { exists, address } = await queryForwardingAccount(
+      config.nobleApi,
+      config.nobleToAgoricChannel,
+      appendedAddr,
+      out,
+      fetch,
+    );
+
+    if (!exists) {
+      nobleSigner ||= await makeSigner(config.nobleSeed, config.nobleRpc, out);
+      const { address: signerAddress, signer } = nobleSigner;
+      try {
+        const res = await registerFwdAccount(
+          signer,
+          signerAddress,
+          config.nobleToAgoricChannel,
+          appendedAddr,
+          out,
+        );
+        out.log(res);
+      } catch (e) {
+        out.error(e);
+        out.error(
+          `Error noble registering forwarding account for ${appendedAddr} on channel ${config.nobleToAgoricChannel}`,
+        );
+        return;
```

`return` suggests failing to register is a success condition. Help me understand that?

It looks like a **silent failure**. It's logged, so it's not entirely silent. But the caller doesn't get the usual signal about success / failure. The unix exit code should be non-zero, right?

---

No you're right, it shouldn't be considered a success condition. I just went back and made sure all these fatal errors actually throw.